drivers/src/soc_ifc.rs

```diff
@@ -365,6 +365,59 @@ impl SocIfc {
             .notif_internal_intr_r()
             .write(|w| w.notif_cmd_avail_sts(true));
     }
+
+    pub fn uds_program_req(&self) -> bool {
+        self.soc_ifc
+            .regs()
+            .ss_dbg_manuf_service_reg_req()
+            .read()
+            .uds_program_req()
+    }
+
+    pub fn set_uds_programming_flow_state(&mut self, in_progress: bool) {
+        self.soc_ifc
+            .regs_mut()
+            .ss_dbg_manuf_service_reg_rsp()
+            .write(|w| w.uds_program_in_progress(in_progress));
+    }
+
+    pub fn set_uds_programming_flow_status(&mut self, flow_succeeded: bool) {
+        self.soc_ifc
+            .regs_mut()
+            .ss_dbg_manuf_service_reg_rsp()
+            .write(|w| {
+                if flow_succeeded {
+                    w.uds_program_success(true);
+                    w.uds_program_fail(false)
+                } else {
+                    w.uds_program_success(false);
+                    w.uds_program_fail(true)
+                }
+            });
+    }
+
+    pub fn uds_seed_dest_base_addr_low(&self) -> u32 {
+        self.soc_ifc.regs().ss_uds_seed_base_addr_l().read()
+    }
```

Comment:
It's a bit strange that the upper part is unused?

nit: maybe also return an AxiAddr type and use .lo inside the UDS programming flow?